types/fx/animations.c

```diff
@@ -0,0 +1,200 @@
+#include <bits/time.h>
+#include <scenefx/types/wlr_scene.h>
+
+#include "scenefx/types/fx/animations.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <wayland-server-core.h>
+#include <wayland-util.h>
+
+static struct fx_animation_manager {
+    struct wl_display *display;
+    struct wlr_scene *scene;
+} manager;
+
+void
+fx_animation_manager_init(struct wl_display *display, struct wlr_scene *scene) {
+    manager.display = display;
+    manager.scene = scene;
+}
+
+struct fx_animation_curve {
+	double params[4];
```

Comment:
It would be nice to know what these 4 parameters are, and what they do as a comment?

https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function/cubic-bezier#parameters